frontend/services/src/recurringEvents/recurringEventsService.ts

```diff
@@ -0,0 +1,39 @@
+import { NewEventData } from "@/interfaces/EventTypes";
+import { NewRecurrenceData, RecurringEventsData, RecurringEventsId } from "@/interfaces/RecurringEventTypes";
+import { Logger } from "@/observability/logger";
+import { collection, doc, Timestamp, writeBatch } from "firebase/firestore";
+import { db } from "../firebase";
+import { CollectionPaths, recurringEventsStatus } from "./recurringEventsConstants";
+import { createEvent } from "../events/eventsService";
+import { durationInDaysCeil } from "../datetimeUtils";
+
+export const recurringEventsServiceLogger = new Logger("recurringEventsServiceLogger");
+
+export async function createRecurringEvents(
+  eventData: NewEventData,
+  recurrenceData: NewRecurrenceData
+): Promise<RecurringEventsId> {
+  recurringEventsServiceLogger.info("createRecurringEvents");
+  const batch = writeBatch(db);
+  const isActive =
+    Timestamp.now().toMillis() < recurrenceData.recurrenceEndDate.toMillis()
+      ? recurringEventsStatus.Active
+      : recurringEventsStatus.Inactive;
+  const docRef = doc(collection(db, CollectionPaths.RecurringEvents, isActive));
+  const recurringEventsData: RecurringEventsData = {
+    eventDataTemplate: eventData,
+    recurrenceData,
+  };
+  batch.set(docRef, recurringEventsData);
+
+  if (
+    isActive === recurringEventsStatus.Active &&
+    durationInDaysCeil(Timestamp.now(), recurrenceData.firstStartDate) <= recurrenceData.createDaysBefore
+  ) {
+    await createEvent(eventData, batch);
```

Comment:
hmm we probably dont want the recurring event engine here right? we want the engine to be in functions and in a cloud cron job.

we want this function to be just the recurrence Template creator, and the actual event creation to still be delegated to the event service. 

No logic for the evaluation of creating an event should be done here. In other words, we don't want to check if we should create the first event here. (the first event will always be created regardless anyways).

Reply:
Ah yes, I agree we should always create first event regardless - this is not schedulling events

Reply:
Yep, the recurring event engine won't be here, but rather in firebase functions - this is still to do